src/operator/nn/pooling-inl.h

```diff
@@ -178,6 +178,11 @@ class PoolingOp {
 template<typename xpu, typename DType>
 PoolingOp<xpu, DType> &GetPoolingOp(const PoolingParam &param) {
   static thread_local PoolingOp<xpu, DType> op;
+  // check if filter size assigned correctly
+  if (param.global_pool == false) {
+    CHECK_GT(param.kernel.ndim(), 0U)
+        << "A positive number must be assigned as filter size";
```

Comment:
Need a better error message.

Reply:
How about "Must assign a positive kernel size"?

Reply:
I think we can use "You need to set the kernel size if global pooling is not used".